securedrop/journalist_app/utils.py

```diff
@@ -201,7 +202,19 @@ def download(zip_basename: str, submissions: List[Union[Submission, Reply]]) ->
     :param list submissions: A list of :class:`models.Submission`s to
                              include in the ZIP-file.
     """
-    zf = current_app.storage.get_bulk_archive(submissions, zip_directory=zip_basename)
+    try:
+        zf = current_app.storage.get_bulk_archive(submissions, zip_directory=zip_basename)
```

Comment:
we can create a follow-up issue to log each missing file. it would be especially helpful if there are missing files across multiple sources.

Reply:
I thinks it should not be too difficult to check for the existence of each requested file in the `get_bulk_archive` method and log the missing files there. We could still raise a `FileNotFoundError` exception if any file is missing at the end of the verification and keep the same handling in the `utils.download` function.

I'll experiment with that and add it to the PR if I have an easy solution. Otherwise, we can raise a follow-up issue

Reply:
Logging of each missing file implemented in 581e91fee5b3723847c09ab1475bc3c8a3abebde.